### PR TITLE
Extend operators exercise with unit-test-like function

### PR DESCRIPTION
**Pull request by hageboeck**
Instead of https://github.com/bernhardmgruber/cpluspluscourse/pull/1

### CODE REVIEW

**Comment by hageboeck** [comment]
I forgot one line, now fixed. The output of the solution looks like this now:
```
Three                                   : PASS    3/1
One third                               : PASS    1/3
One third times two                     : PASS    2/3
One third times two                     : PASS    2/3
Three times one third                   : PASS    1/1
Three times one third                   : PASS    1/1
One third times two                     : PASS    2/3
One third times one third               : PASS    1/9
One third times itself twice            : PASS    1/81
One third plus 2                        : PASS    7/3
2 plus one third                        : PASS    7/3
One third plus two sixth                : PASS    2/3
One third plus one and added to itself  : PASS    8/3
One third == one third                  : PASS    true
One third != one forth                  : PASS    true
One third == two sixth                  : PASS    true
One third != three sixth                : PASS    true
athird < athird                         : PASS    false
afourth < athird                        : PASS    true
athird  <= athird                       : PASS    true
athird  <= afourth                      : PASS    false
athird  > athird                        : PASS    false
afourth > athird                        : PASS    false
athird >= athird                        : PASS    true
athird >= afourth                       : PASS    true
```

Including the bug that was in the alpha version of the exercise, this would have looked like below.
I think that proves the point the this test-like infrastructure is useful. Please check that I didn't hardcode any wrong solution!!! 🙂 
```
c++ -g -std=c++17 -O0 -Wall -Wextra -L. -o operators.sol solution/operators.sol.cpp
Three                                   : PASS    3/1
One third                               : PASS    1/3
One third times two                     : ** FAIL **    1/3
One third times two                     : ** FAIL **    1/3
Three times one third                   : PASS    1/1
Three times one third                   : ** FAIL **    1/3
One third times two                     : ** FAIL **    1/3
One third times one third               : PASS    1/9
One third times itself twice            : PASS    1/81
One third plus 2                        : PASS    7/3
2 plus one third                        : PASS    7/3
One third plus two sixth                : PASS    2/3
One third plus one and added to itself  : PASS    8/3
One third == one third                  : PASS    true
One third != one forth                  : PASS    true
One third == two sixth                  : PASS    true
One third != three sixth                : PASS    true
athird < athird                         : PASS    false
afourth < athird                        : PASS    true
athird  <= athird                       : PASS    true
athird  <= afourth                      : PASS    false
athird  > athird                        : PASS    false
afourth > athird                        : PASS    false
athird >= athird                        : PASS    true
athird >= afourth                       : PASS    true
```

**Comment by sponce** [comment]
All fine for me. If @bernhardmgruber agrees, let's merge

File: code/operators/operators.cpp [hunk 1]
```diff
@@ -7,52 +7,62 @@ class Fraction {
...
 int main() {
   // create a fraction with values 3 (which is 3/1) and 1/3
   const Fraction three{3};
   const Fraction athird{1, 3};
 
   // print the fractions
-  std::cout << "Three: " << three << '\n';
-  std::cout << "One third: " << athird << '\n';
+  printAndCheck("Three", three, Fraction{3, 1});
+  printAndCheck("One third", athird, Fraction{1, 3});
```
Comment (bernhardmgruber) [inline code/operators/operators.cpp:27]: Please leave the first two `cout` statements please, so students know better that this is the first operator to implement.

File: code/operators/operators.cpp [hunk 2]
```diff
@@ -61,31 +71,28 @@ int main() {
   // add in place
   f = athird;
   f += f += 1;
-  std::cout << "One third plus one and added to itself: " << f << '\n';
+  printAndCheck("One third plus one and added to itself", f, Fraction{8, 3});
   // again, try to refactor and implement operator+ in terms of operator+=
 
   std::cout << std::boolalpha;  // print bools as 'true' or 'false' from now on
 
   // compare two fractions for equality, all of these should be 'true'
```
Comment (bernhardmgruber) [inline code/operators/operators.cpp:79]:
```suggestion
  // more equality comparisons
```
We can drop the fact that those should return true. And students must now implement `opeartor==` sooner.

**Review by bernhardmgruber — CHANGES_REQUESTED** [review]

**Comment by hageboeck** [comment]
All suggestions taken care of, @bernhardmgruber. Thanks.

**Pull request merged**